### DISCUSSION TITLE
Windows Build Size

### DISCUSSION THREAD

**Discussion by jonassorgenfrei** [AcademySoftwareFoundation/OpenRV#65] [Build Issues] [answered]
Heyhey

i succesfully build openrv, but i'm wondering why the size of the app folder is about 4.45 GB.

I was following the build instructions (for the release version). 
But it looks like there where debug files build esp. for the dependencies. 

![grafik](https://user-images.githubusercontent.com/34287983/217695802-db0436de-ca73-4d51-bf17-61d9e550271e.png)

Do i have to do anything else to make sure only the release is build? 

Thanks for your help in advance :)

**Comment by rogernelson** [comment]
Hi @jonassorgenfrei.  You can get aRelease build with the ` -DCMAKE_BUILD_TYPE=Release` flag during the configure stage.  For Windows, you'll also need to add the `--config Release` flag to the build stage.

If you like, you can also use the aliases to do it for you.  If you do `source rvcmds.sh` then you can simply do `rvmk` to get a release build or `rvmkd` for a debug build.

Hope that helps!

**Comment by bernie-laberge** [comment]
Hello @jonassorgenfrei,

In addition to what Roger has just mentioned, you can do an extra step, the install step, that will prune the pdb files from your build. You can invoke the build step with the `rvinst` alias that comes when you do `source rvcmds.sh`.

Once you've run the install step, you should end up with an RV taking about 1.5 GB of disk space.
We haven't really optimized it for space. Sorry about that.

I hope this helps.

**Comment by jonassorgenfrei** [comment]
Hmm I'm pretty sure i copied the commands just like this from the docs: 

```
cmake -B cmake-build -G "Visual Studio 16 2019" -A x64 -DCMAKE_BUILD_TYPE=Release -DRV_DEPS_WIN_PERL_ROOT=/c/Strawberry/perl/bin -DRV_DEPS_QT5_LOCATION=/d/customlibs/Qt/5.15.2/msvc2019_64
cmake --build cmake-build --config Release --target rv -j8
cmake --install cmake-build --config Release
```
But may be accidentally deleted an important part of the command; will double check but thanks for the help will let you know if this fixed and removed the debug. 


Or is it correct that the folder in cmake-build/stage/app is that big and contains the pdb files even during release build?

**Comment by geffrak** [comment]
The stage/app folder contains EVERYTHINGS, so you can easily debug stuff. You need to run the install step to copy it into a final folder with unrequired files filtered out.

**Comment by jonassorgenfrei** [comment]
I see that makes sense allright will give it a try. Thank you for your help ☺️

**Comment by geffrak** [comment, marked as answer]
I just noticed (thanks!) that the docs don't mention about the install step properly.
`cmake --install cmake-build --config Release --prefix /Absolute/path/to/final/location/RV`

**Comment by jonassorgenfrei** [comment]
Hey @geffrak thanks for your help this was exactly the thing i missed. 
I expected thtat the stage/app folder when building as release just contains the release build but was wrong there. 

With installing using the command you provided i get a build size of around 1.5 gig.
